src/core/util/bundler.ts

```diff
@@ -1,57 +1,118 @@
   const cmpAnnotation: ComponentMetadata = annotations[ 0 ];
   const { directives = [], pipes = [], providers = [], viewProviders = [] }={} = cmpAnnotation;
 
   // process component
   const [cmpName,cmpFactoryFn] = provide( ComponentClass );
-  const { providerName, providerMethod, moduleMethod } = _getNgModuleMetadataByType( ComponentClass );
+  const { providerName, providerMethod, moduleMethod } = _getAngular1ModuleMetadataByType( ComponentClass );
 
-  if ( _isTypeRegistered( cmpName, ngModule, providerName, providerMethod ) ) {
-    return ngModule;
+  if ( _isTypeRegistered( cmpName, angular1Module, providerName, providerMethod ) ) {
+    return angular1Module;
   }
 
   // @TODO register via this once requires are resolved for 3 types of attr directive from template
-  // _registerTypeProvider( ngModule, ComponentClass, { moduleMethod, name: cmpName, value: cmpFactoryFn } );
-  ngModule[moduleMethod]( cmpName, cmpFactoryFn );
+  // _registerTypeProvider( angular1Module, ComponentClass, { moduleMethod, name: cmpName, value: cmpFactoryFn } );
+  angular1Module[moduleMethod]( cmpName, cmpFactoryFn );
 
   // 1. process component/directive decorator providers/viewProviders/pipes
-  _normalizeProviders( ngModule, providers );
-  _normalizeProviders( ngModule, viewProviders );
-  _normalizeProviders( ngModule, pipes );
+  _normalizeProviders( angular1Module, providers );
+  _normalizeProviders( angular1Module, viewProviders );
+  _normalizeProviders( angular1Module, pipes );
 
 
   // step through all directives
   ListWrapper.flattenDeep(directives).forEach( ( directiveType: Type ) => {
-    bundle( directiveType, [], ngModule );
+    _bundleComponent( directiveType, [], angular1Module );
   } );
 
   // 2. process otherProviders argument
-  // - providers can be string(ngModule reference), Type, StringMap(providerLiteral)
+  // - providers can be string(angular1Module reference), Type, StringMap(providerLiteral)
   // - directives can't be registered as via global providers only @Injectable,@Pipe,{provide:any,use*:any}
-  // registerProviders(ngModule, otherProviders);
-  _normalizeProviders( ngModule, otherProviders );
+  // registerProviders(angular1Module, otherProviders);
+  _normalizeProviders( angular1Module, otherProviders );
+
+  return angular1Module;
+}
+
+export function bundle( NgModuleClass: Type, otherProviders: any[] = [], existingAngular1Module?: ng.IModule ): ng.IModule {
+
+  const angular1ModuleName = getInjectableName( NgModuleClass );
+  const angular1Module = existingAngular1Module || global.angular.module( angular1ModuleName, [] );
+  const annotations = reflector.annotations( NgModuleClass );
+  const ngModuleAnnotation: NgModuleMetadata = annotations[ 0 ];
+  if (!isNgModule(ngModuleAnnotation)) {
+    throw new Error(`bundle() requires an @NgModule as it's first argument`)
+  }
+  const { declarations = [], providers = [], imports = [] }={} = ngModuleAnnotation;
```

Comment:
oh this comment was supposed to be in other place. github acting weird I guess ? :D